src/main/java/org/sonarlint/intellij/notifications/SonarLintProjectNotifications.kt

```diff
@@ -0,0 +1,331 @@
+import org.sonarsource.sonarlint.core.commons.log.ClientLogOutput
+import java.util.Arrays
+import java.util.stream.Stream
+
+
+@Service(Service.Level.PROJECT)
+class SonarLintProjectNotifications(private val myProject: Project) {
+
+    companion object {
+        val IN_CONTEXT_PROMOTION_GROUP: NotificationGroup = NotificationGroupManager.getInstance().getNotificationGroup("SonarLint: In Context Promotions")
+        val BINDING_PROBLEM_GROUP: NotificationGroup = NotificationGroupManager.getInstance().getNotificationGroup("SonarLint: Server Binding Errors")
+        val SERVER_NOTIFICATIONS_GROUP: NotificationGroup = NotificationGroupManager.getInstance().getNotificationGroup("SonarLint: Server Notifications")
+        val BINDING_SUGGESTION_GROUP: NotificationGroup = NotificationGroupManager.getInstance().getNotificationGroup("SonarLint: Binding Suggestions")
+        val OPEN_IN_IDE_GROUP: NotificationGroup = NotificationGroupManager.getInstance().getNotificationGroup("SonarLint: Open in IDE")
+        val SECRET_DETECTION_GROUP: NotificationGroup = NotificationGroupManager.getInstance().getNotificationGroup("SonarLint: Secrets detection")
+        val TAINT_GROUP: NotificationGroup = NotificationGroupManager.getInstance().getNotificationGroup("SonarLint: Taint vulnerabilities")
+        val ANALYZER_REQUIREMENT_GROUP: NotificationGroup = NotificationGroupManager.getInstance().getNotificationGroup("SonarLint: Analyzer Requirement")
+        val ISSUE_RESOLVED_GROUP: NotificationGroup = NotificationGroupManager.getInstance().getNotificationGroup("SonarLint: Mark Issue as Resolved")
+        val HOTSPOT_REVIEW_GROUP: NotificationGroup = NotificationGroupManager.getInstance().getNotificationGroup("SonarLint: Security Hotspot Review")
+        val SONARLINT_GROUP: NotificationGroup = NotificationGroupManager.getInstance().getNotificationGroup("SonarLint")
+
+        private const val UPDATE_BINDING_MSG = "\n<br>Please check the SonarLint project configuration"
+        private const val TITLE_SONARLINT_INVALID_BINDING = "<b>SonarLint - Invalid binding</b>"
+        private const val TITLE_SONARLINT_SUGGESTIONS = "<b>SonarLint suggestions</b>"
+
+        fun get(project: Project): SonarLintProjectNotifications {
+            return SonarLintUtils.getService(project, SonarLintProjectNotifications::class.java)
+        }
+
+        fun projectLessNotification(title: String?, message: String, type: NotificationType, action: AnAction? = null): Notification {
+            return SONARLINT_GROUP.createNotification(
+                title ?: "",
+                message,
+                type
+            ).apply {
+                isImportant = type != NotificationType.INFORMATION
+                icon = SonarLintIcons.SONARLINT
+                action?.let { addAction(it) }
+                notify(null)
+            }
+        }
+    }
+
+    @Volatile
+    private var shown = false
```

Comment:
To me, this is true if any notification is shown, while it's just a few of them. Maybe we could find a better name, like `bindingErrorNotificationShown`?

Reply:
Actually, this one is not for the binding but for the storage, but I will rename it 👍🏻